### PR TITLE
Check track name changes before modify-cluster

### PR DESCRIPTION
**Pull request by mashuaihh**
Before this fix, if cx doesn't provide a track name in the CFN template, we'll call the create-cluster w/o the track name, and create-cluster by default creats a cluster w/ "current" track. When cx does add "current" track to the template, our code thinks it needs an update and calls modify-cluster w/ "current" track, and modify-cluster throws exception that the cluster is already on "current" track.

We should have handled this and not call modify-cluster when track name changes from null to "current".

This commit checks the previousModel's track setup and calls modify-cluster accordingly

Tested locally in Docker, previously failing modify-cluster is now passing. The code also handles when modify-cluster needs to be called but track doesn't need to be modified

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: aws-redshift-cluster/src/test/java/software/amazon/redshift/cluster/UpdateHandlerTest.java [hunk 1]
```diff
@@ -869,4 +851,79 @@ public void testModifyClusterAttributes(
...
+
+    @ParameterizedTest
+    @MethodSource("maintenanceTrackTestData")
+    public void testMaintenanceTrackModifyLogic(
+            String beforeTrack,
+            String afterTrack,
+            boolean shouldCallModifyCluster,
+            String expectedModifyTrackName
+    ) {
+        ResourceModel previousModel = BASIC_MODEL.toBuilder().publiclyAccessible(true).maintenanceTrackName(beforeTrack).build();
```
Comment (hkondru) [inline aws-redshift-cluster/src/test/java/software/amazon/redshift/cluster/UpdateHandlerTest.java:882]: Is this intended to have publicly Accessible clusters? if yes, why?

Comment (mashuaihh) [reply aws-redshift-cluster/src/test/java/software/amazon/redshift/cluster/UpdateHandlerTest.java:882]: in this unit test, I'm setting this to trigger the modify-cluster call. We need this extra unit test because in case we need to call modify-cluster but w/o any track changes, we need to pass in "null" to avoid modify-cluster failures

**Review by hkondru — APPROVED** [review]

**Review by bowliang — APPROVED** [review]

**Review by vramanet — APPROVED** [review]

File: aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java [hunk 2]
```diff
@@ -690,6 +693,44 @@ static String finalClusterSnapshotIdentifierBuilder(String clusterIdentifier, bo
...
   }
 
+  // if we don't provide any track name when we call create-cluster API,
+  // the default track name is "current"
+  // when cx updates the CFN template to include a track name, "current",
+  // we used to call modify-cluster with track("current"),
+  // and we'd get exception "track is already on 'current'".
+  // The following check is to avoid this unnecessary update call
+  static boolean shouldModifyMaintenanceTrack(final ResourceModel oldModel, final ResourceModel newModel) {
+    final String oldTrackName = oldModel.getMaintenanceTrackName();
```
Comment (vramanet) [inline aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java:703]: oldTrackName is never null. Its either 'current' or whatever the user last provided trackName

Comment (mashuaihh) [reply aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java:703]: the model can stay null (the API response is not), that's why we had the issue in the first place

Comment (vramanet) [reply aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java:703]: https://code.amazon.com/packages/AWSCloudFormationResourceProvidersRedshift/blobs/55a93e83aa0674357e4707ea3d44e7e17b1b5cd9/--/aws-redshift-cluster/src/main/java/software/amazon/redshift/cluster/Translator.java#L646
